File: 프로그래머스/lv2/64065. 튜플/튜플.py
def solution(s):
    ans = []
    set_ans = set()
    s = s[2:-2].split('},{')
    s.sort(key=lambda x: len(x))

    for i in range(len(s)):
        temp = s[i].split(',')
        for j in temp:
            if int(j) not in set_ans:
                ans.append(int(j))
                set_ans.add(int(j))

    return ans

File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        HashSet<Integer> check = new HashSet<>();
        List<Integer> result = new ArrayList<>();
        String str = s.substring(2, s.length() - 2);

        String[] strArr = str.split("\\},\\{");
        Arrays.sort(strArr, ((o1, o2) -> o1.length() - o2.length()));

        for (String tuple : strArr) {
            String[] newTuple = tuple.split(",");
            for (String num : newTuple) {
                int intNum = Integer.parseInt(num);
                if (!check.contains(intNum)) {
                    result.add(intNum);
                    check.add(intNum);
                }
            }
        }

        return result.stream().mapToInt(i -> i).toArray();
    }
}